resources/assets/js/topics.js

```diff
@@ -0,0 +1,72 @@
+let topics = null;
+
+export default {
+    all() {
+        if (topics !== null) {
+            return new Promise(function (resolve) {
+                resolve(topics);
+            });
+        }
+
+        return new Promise(function (resolve, reject) {
+            Vue.http.get('topics').then(
+                response => {
+                    topics = response.data;
+                    resolve(topics);
+                },
+                response => {
+                    reject();
+                    console.log('error', response);
+                }
+            );
+        });
+    },
+
+    add(topic) {
+        return new Promise( (resolve, reject) => {
```

Comment:
Why is this wrapped in a promise but the others aren't? (I don't know much about promises tbh)